src/website/webview/views.py

```diff
@@ -500,8 +501,40 @@ def get_context_data(self, **kwargs: Any) -> Any:
 
         context["status_filter"] = self.status_filter
 
+        all_affected_pk = list()
         for obj in context["object_list"]:
             obj.packages = channel_structure(obj.derivations.all())
+            # We cache the list of AffectedProduct ids per suggestion for later
+            obj.affected_pk = obj.cve.container.values_list("affected", flat=True)
+            all_affected_pk.extend(obj.affected_pk)
+        affected = AffectedProduct.objects.prefetch_related(
+            Prefetch("versions", to_attr="all_versions"),
+            Prefetch("cpes", to_attr="all_cpes"),
+        ).in_bulk(id_list=all_affected_pk)
+
+        for obj in context["object_list"]:
+            obj.affected_packages = dict()
+            for pk in obj.affected_pk:
+                if pk is not None:
+                    a = affected[pk]
+                    if a.package_name:
+                        if a.package_name not in obj.affected_packages:
+                            obj.affected_packages[a.package_name] = {
+                                "version_constraints": set(),
+                                "cpes": set(),
+                            }
+                        obj.affected_packages[a.package_name][
+                            "version_constraints"
+                        ].update(
+                            [
+                                (vc.status, vc.version_constraint_str())
+                                for vc in a.all_versions
```

Comment:
Using `a.all_versions` like this runs fine, but produces a pyright error, as it apparently doesn't understand how django adds fields via prefetch:

```
/home/kerstin/git/nix-security-tracker/src/website/webview/views.py                                                                    
  /home/kerstin/git/nix-security-tracker/src/website/webview/views.py:531:45 - error: Cannot access attribute "all_versions" for class 
"AffectedProduct"                                                                                                                      
    Attribute "all_versions" is unknown (reportAttributeAccessIssue)
  /home/kerstin/git/nix-security-tracker/src/website/webview/views.py:535:52 - error: Cannot access attribute "all_cpes" for class "Aff
ectedProduct"
    Attribute "all_cpes" is unknown (reportAttributeAccessIssue)
2 errors, 0 warnings, 0 informations
```

I've had this problem in the past, but forgot how/wether I solved it. Just not using `to_attr` doesn't help, as then I can't iterate on the field at runtime.

Reply:
Figured this out. The trick is to not use `to_attr` and then use `.all()` on the field.